src/object/obj_tx.c

```diff
@@ -2190,7 +2190,10 @@ dc_tx_add_update(struct dc_tx *tx, daos_handle_t oh, uint64_t flags,
 	dcsr->dcsr_opc = DCSO_UPDATE;
 	dcsr->dcsr_nr = nr;
 	dcsr->dcsr_dkey_hash = obj_dkey2hash(obj->cob_md.omd_id, dkey);
-	dcsr->dcsr_api_flags = flags;
+	dcsr->dcsr_api_flags = flags & ~(DAOS_COND_DKEY_INSERT |
+					 DAOS_COND_DKEY_UPDATE |
+					 DAOS_COND_AKEY_INSERT |
+					 DAOS_COND_AKEY_UPDATE);
```

Comment:
Might want to add a comment to daos_obj.h pointing back here, so that if new conditional flags are added there, the code here will also be updated. Or, perhaps we could simply "& ~(DAOS_COND_MASK)" for all cases (here and below).

Reply:
Yes, I refreshed the patch as suggested.